spec/features/user/connect_social_account_spec.rb

```diff
@@ -0,0 +1,60 @@
+require "rails_helper"
+
+feature "Connect social account" do
+  context "oauth confirmed" do
+    include_context :stub_omniauth
+
+    before { click_connect_fb }
+
+    context "user confirmed" do
+      let!(:user) { create(:user, :confirmed, :from_auth_hashie) }
+
+      scenario "User connects social account" do
+        expect(page).to have_connected_account("Facebook")
+      end
+    end
+
+    context "user not confirmed" do
+      let!(:user) { create(:user, :from_auth_hashie) }
+
+      scenario "User have to confirm own account" do
+        expect(page).to have_connected_account("Facebook")
+
+        open_email(user.email)
+
+        expect(current_email).to have_subject("Confirmation instructions")
+        expect(current_email).to have_body_text(user.full_name)
+      end
+    end
+  end
+
+  context "oauth not confirmed" do
+    include_context :stub_not_verified_omniauth
+
+    before { click_connect_fb }
+
+    context "user confirmed" do
+      let!(:user) { create(:user, :confirmed, :from_auth_hashie) }
+
+      scenario "User connects social account" do
+        expect(page).to have_connected_account("Facebook")
+      end
+    end
+
+    context "user not confirmed" do
+      let!(:user) { create(:user, :from_auth_hashie) }
+
+      scenario "User sees alert" do
+        expect(page).not_to have_css(".js-social-profiles")
```

Comment:
In this page can be  2 "Facebook" words - in the prompt section and in the connected profiles list (".js-social-profiles"). And for this reason I check css here. Anyway thanks, I'll try replace it with more robust solution.